Tests/BSONKitTests/BinaryDecodableValueTests.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  BinaryDecodableValueTests.swift
+//
+//
+//  Created by Christopher Richez on September 27th, 2021
+//
+
+import BSONCore
+import XCTest
+
+/// The test case for the `BinaryDecodableValue` protocol.
+///
+/// This case tests `Array<UInt8>` conformance as a proof of concept.
+/// The test suite only includes initialization from a full generic BSON binary value.
+class BinaryDecodableValueTests: XCTestCase {
+    func testByteArray() throws {
+        /// The document from the start of the binary value
```

Comment:
Add details on the contents of the document to decode.
```suggestion
        /// The document from the start of the binary value. Includes:
        /// - 4 size bytes
        /// - 1 subtype byte
        /// - 4 value bytes
        /// - 4 bytes past the end of the value
```